server/models/tale.py

```diff
@@ -146,3 +146,47 @@ def updateTale(self, tale):
         """
         tale['updated'] = datetime.datetime.utcnow()
         return self.save(tale)
+
+    def setAccessList(self, doc, access, save=False, user=None, force=False,
+                      setPublic=None, publicFlags=None):
+        """
+        Overrides AccessControlledModel.setAccessList to encapsulate ACL
+        functionality for a tale.
+
+        :param doc: the tale to set access settings on
+        :type doc: girder.models.tale
+        :param access: The access control list
+        :type access: dict
+        :param save: Whether the changes should be saved to the database
+        :type save: bool
+        :param user: The current user
+        :param force: Set this to True to set the flags regardless of the passed in
+            user's permissions.
+        :type force: bool
+        :param setPublic: Pass this if you wish to set the public flag on the
+            resources being updated.
+        :type setPublic: bool or None
+        :param publicFlags: Pass this if you wish to set the public flag list on
+            resources being updated.
+        :type publicFlags: flag identifier str, or list/set/tuple of them,
+            or None
+        """
+        if setPublic is not None:
+            self.setPublic(doc, setPublic, save=False)
+
+        if publicFlags is not None:
+            doc = self.setPublicFlags(doc, publicFlags, user=user, save=False,
+                                      force=force)
+
+        doc = AccessControlledModel.setAccessList(self, doc, access,
+                                                  user=user, save=save, force=force)
+
+        image = AccessControlledModel.model('image', 'wholetale').load(
+            doc['imageId'], user=user, level=AccessType.ADMIN)
+
+        if image:
+            AccessControlledModel.model('image', 'wholetale').setAccessList(
+                image, access, user=user, save=save, force=force,
+                setPublic=setPublic, publicFlags=publicFlags)
+
```

Comment:
What about folder associated with the Tale?

Reply:
Woops, somehow forgot about the folder. I'll update the PR to propogate the ACL to the folder too.